packages/charts/src/vaadin-chart.js

```diff
@@ -69,6 +70,28 @@ export function deepMerge(target, source) {
   /* eslint-enable no-invalid-this */
 });
 
+// Monkeypatch the onDocumentMouseMove method to fix the check for the source of the event
+// Due to the fact that the event is attached to the document, the target of the event is
+// the <vaadin-chart> element, so we need to use the composedPath to get the actual target (#7107)
+Pointer.prototype.onDocumentMouseMove = function (e) {
```

Comment:
```js
const originalOnDocumentMouseMove = Pointer.prototype.onDocumentMouseMove;
Pointer.prototype.onDocumentMouseMove = function (e) {
  const newTarget = e.composedPath()[0];
  Object.defineProperty(e, 'target', { writable: false, value: newTarget });
  return originalOnDocumentMouseMove.call(this, e);
};
```

Better or worse?

Reply:
Surely this has less LOC, but I am not sure which one is better. Both are relying on how HC works on this version, so any HC version update would require us to revisit this and adjust (or even remove) this code entirely.